src/transactions/TransactionUtils.h

```diff
@@ -262,7 +262,8 @@ bool accountFlagMaskCheckIsValid(uint32_t flag, uint32_t ledgerVersion);
 bool hasMuxedAccount(TransactionEnvelope const& e);
 
 bool
-isTransactionXDRValidForCurrentProtocol(AppConnector& app,
+isTransactionXDRValidForCurrentProtocol(uint32_t currProtocol,
```

Comment:
nit: This function should probably be renamed to just `isTransactionXDRValidForProtocol` (as we pass the protocol explicitly).

Also it seems like we can pass `Config` instead of the whole `AppConnector` now